org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/XtextBuilder.java

```diff
@@ -373,5 +380,17 @@ protected void doClean(ToBeBuilt toBeBuilt, IProgressMonitor monitor) throws Cor
 			progress.worked(1);
 		}
 	}
-	
+
+	@Override
+	public ISchedulingRule getRule(int kind, Map<String, String> args) {
+		switch (preferenceAccess.getSchedulingRule(getProject())) {
+			case WORKSPACE: return getProject().getWorkspace().getRoot();
```

Comment:
Super experimental would be the 'null' rule.

Reply:
Why not? When we are already at it, it makes sense to add this.